host/host.go

```diff
@@ -371,9 +402,19 @@ func (hn *hostImpl) Router(h host.Host) (routing.PeerRouting, error) {
 	return hn.IpfsDHT, nil
 }
 
+// PubSub returns the host node PubSub Function
+func (hn *hostImpl) Pubsub() *ps.PubSub {
+	return hn.PubSub
+}
+
+// Routing returns the host node Peer Routing Function
+func (hn *hostImpl) Routing() routing.Routing {
```

Comment:
Should we be returning a pointer to a router?

Reply:
This is being used in the Matrix Module for finding peers utilizing the nodes DHT.